src/track/track.cpp

```diff
@@ -1108,7 +1108,16 @@ bool Track::importPendingCueInfosWhileLocked() {
             m_record.getMetadata().getStreamInfo().getSignalInfo().getSampleRate());
     const auto trackId = m_record.getId();
     QList<CuePointer> cuePoints;
-    cuePoints.reserve(m_pCueInfoImporterPending->size());
+    cuePoints.reserve(m_pCueInfoImporterPending->size() + m_cuePoints.size());
+
+    // Avoid unnecessary data loss by not discarding/overwriting existing cues
```

Comment:
The comment is not easy to understand. How about "Preserve all existing cues with types that are not available for importing."